log4j-core-test/src/test/java/org/apache/logging/log4j/core/config/plugins/processor/PluginCacheTest.java

```diff
@@ -48,9 +42,51 @@ public void testOutputIsReproducibleWhenInputOrderingChanges() throws IOExceptio
         assertArrayEquals(cacheData(cacheA), cacheData(cacheB));
     }
 
-    private void createCategory(final PluginCache cache, final String categoryName, final List<String> entryNames) {
+    @Test
+    void testDeserialize() throws IOException {
+        final PluginCache expected = createSampleCache();
+
+        final Map<String, Map<String, PluginEntry>> actual = new TreeMap<>();
+        PluginCache.loadInputStream(new ByteArrayInputStream(cacheData(expected)), actual);
+
+        assertThat(actual).as("Deserialized plugin cache").isEqualTo(expected.getAllCategories());
+    }
+
+    @Test
+    void testConcatenationOfCaches() throws IOException {
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
+        boolean first = true;
+        for (final String categoryName : Arrays.asList("one", "two")) {
+            final PluginCache cache = new PluginCache();
+            createCategory(cache, categoryName, Arrays.asList("alpha", "bravo", "charlie"));
+            cache.writeCache(output);
+            if (first) {
+                output.write('\n');
+                first = false;
+            }
+        }
+
+        final Map<String, Map<String, PluginEntry>> actual = new TreeMap<>();
+        PluginCache.loadInputStream(new ByteArrayInputStream(output.toByteArray()), actual);
+
+        assertThat(actual)
+                .as("Deserialized plugin cache")
+                .isEqualTo(createSampleCache().getAllCategories());
```

Comment:
_Nit:_ I'd prefer `createSampleCache()` accept categories and entries, instead of assuming hardcoded values will be identical in multiple places.